dht/dht.go

```diff
@@ -208,8 +238,8 @@ func (nid *nodeID) ByteString() string {
 	return string(buf[:])
 }
 
-type node struct {
-	addr          dHTAddr
+type Node struct {
```

Comment:
Still looking into this while I resolve the rest. Without creating a new object with a guaranteed interface that carries "data hooks can use", exposing this and its attributes is the cleanest way I can see to give hook authors access to useful data?